script.js

```diff
@@ -1,4 +1,184 @@
+
+// Dice roll function to get random dice number
+var diceRollResult = function () {
+  var rollResult = Math.floor(Math.random() * 6) + 1;
+  return rollResult;
+};
+
+// Evaluate results
+var evaluateResults = function (input) {
+  //For both player first roll
+  if ((turnOrOrder == 0 || turnOrOrder == 2) && whichDiceOrder == 0) {
+    whichDiceOrder = 1;
+    turnOrOrder += 1;
+    playerThrow[0] = diceRollResult();
+    playerThrow[1] = diceRollResult();
+    myOutputValue = `You rolled ${playerThrow[0]} and ${playerThrow[1]}.`;
+    console.log(turnOrOrder);
+
+    //If both die result is the same, skip order selection
+    if (playerThrow[0] == playerThrow[1]) {
+      whichDiceOrder = 0;
+      turnOrOrder += 1;
+      //result for P1
+      if (turnOrOrder == 2) {
+        pOneResult = "" + playerThrow[0] + playerThrow[1];
+        //result for P2
+        myOutputValue =
+          myOutputValue +
+          `<br><br>Your hand is ${playerThrow[0]}${playerThrow[1]}.`;
+        return myOutputValue;
+      } else if (turnOrOrder == 4) {
+        pTwoResult = "" + playerThrow[0] + playerThrow[1];
+        myOutputValue =
+          myOutputValue +
+          `<br><br>Your hand is ${playerThrow[0]}${playerThrow[1]}.`;
+        return myOutputValue;
+      }
+    }
+
+    //Else ask player which order they want to choose
+    myOutputValue =
+      myOutputValue +
+      ` <br><br>Which order would you like to put them?<br> 1) die 1 first : ${playerThrow[0]}${playerThrow[1]} <br> 2) die 2 first : ${playerThrow[1]}${playerThrow[0]}`;
+    return myOutputValue;
+  } // Player to choose order of die:
+  else if (turnOrOrder == 1 && whichDiceOrder == 1) {
+    // Player 1 return message (die 1 first)
+    if (input == 1) {
+      turnOrOrder += 1;
+      whichDiceOrder = 0;
+      pOneResult = "" + playerThrow[0] + playerThrow[1];
+      return `Ok, your hand is ${pOneResult}. <br><br>It is now Player 2's turn.`;
+    } // player 1 return message (die 2 first)
+    else if (input == 2) {
+      turnOrOrder += 1;
+      whichDiceOrder = 0;
+      pOneResult = "" + playerThrow[1] + playerThrow[0];
+      return `Ok, your hand is ${pOneResult}. <br><br>It is now Player 2's turn.`;
+    } else {
+      return `Please select a valid number.<br><br>${myOutputValue}`;
+    }
+  } // Player 2 choose order of die
+  else if (turnOrOrder == 3 && whichDiceOrder == 1) {
+    if (input == 1) {
+      console.log("p2 1");
+      turnOrOrder = 4;
+      whichDiceOrder = 0;
+      pTwoResult = "" + playerThrow[0] + playerThrow[1];
+      myOutputValue = `Ok, your hand is ${pTwoResult}.`;
+      return myOutputValue;
+    } else if (input == 2) {
+      console.log("p2 2");
+      turnOrOrder = 4;
+      whichDiceOrder = 0;
+      pTwoResult = "" + playerThrow[1] + playerThrow[0];
+      myOutputValue = `Ok, your hand is ${pTwoResult}`;
+      return myOutputValue;
+    } else {
+      return `Please select a valid number.<br><br>${myOutputValue}`;
+    }
```

Comment:
same as above, we are making use of early return so no need to nest if / else if / else

Reply:
i would also put the negative case up top first and check for that

```
    if (input != 1 && input != 2) {
      return `Please select a valid number.<br><br>${myOutputValue}`;
    }

    turnOrOrder = 4;
    whichDiceOrder = 0;

    if (input == 1) {
      console.log("p2 1");
      pTwoResult = "" + playerThrow[0] + playerThrow[1];
      myOutputValue = `Ok, your hand is ${pTwoResult}.`;
      return myOutputValue;
    } 
    
    console.log("p2 2");
    pTwoResult = "" + playerThrow[1] + playerThrow[0];
    myOutputValue = `Ok, your hand is ${pTwoResult}`;
    return myOutputValue;
```

Reply:
but good input validation here and i like that you display the 'please select a valid number' message to the user along with the dice rolls again! a comment i have for many students is that when they output this msg, they don't show the rolls again and it is very possible the user forgot the dice rolls